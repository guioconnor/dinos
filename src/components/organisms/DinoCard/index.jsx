import React from "react";
import styled from "styled-components";
import DietIcon from "../../atoms/DietIcon";
import HabitatIcon from "../../atoms/HabitatIcon";
import DinoImage from "../../atoms/DinoImage";
import PeriodGraph from "../../atoms/PeriodGraph";

const CardWrapper = styled.div`
  width: ${props => props.width ? `${props.width}px` : '100%'};
  min-height: ${props => props.width ? `${props.width}px` : '100%'};
  opacity: ${props => `${props.opacity}`};
  display: flex;
  flex-direction: column;
  background: rgba(255, 255, 255, .8);
  color: #333;
  padding: 10%;
  border-radius: 10px;
  text-decoration: none;
  box-shadow: 3px 3px 3px rgba(0, 0, 0, 0.2);

  &:hover {
    background: #eee;
  }
`;

const ElementContainer = styled.div`
  width: 100px;
  height: 100px;
  padding: 5%;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

const Icons = styled.ul`
  display: flex;
  justify-content: center;

  & > span {
    margin: 0 5px;
  }
`;

const H2 = styled.h2`margin-top: 20px;`;

const DinoCard = ({
  item,
  displayDetails,
  displayImage = true,
  opacity = 1,
  width = null,
  prefix,
}) => (
    <CardWrapper width={width} opacity={opacity}>
      {displayDetails && <H2>{item.name}</H2>}
      {displayImage && (
        <ElementContainer>
          <DinoImage name={item.itemId} prefix={prefix} alt={item.name} width={50} height={50} silhouette={item.silhouette} />
        </ElementContainer>
      )}
      {displayDetails && (
        <div>
          <Icons>
            <DietIcon diet={item.diet} />
            <HabitatIcon habitat={item.habitat} />
          </Icons>
          <ElementContainer>
            {item.mya && <PeriodGraph mya={item.mya} simple />}
          </ElementContainer>
        </div>
      )}
    </CardWrapper>
  );

export default DinoCard;
