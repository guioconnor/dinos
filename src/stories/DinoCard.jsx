import React from "react";
import { storiesOf } from "@storybook/react";
import DinoCard from "../components/organisms/DinoCard";

const singleDinoData = {
  name: "name",
  image: "image",
  pronunciation: "pronunciation",
  meaningOfName: "meaningOfName",
  diet: "diet",
  length: "length",
  period: "period",
  mya: "mya",
  info: "info"
};

storiesOf("DinoCard", module).add(
  "default",
  <DinoCard item={singleDinoData} />
);
