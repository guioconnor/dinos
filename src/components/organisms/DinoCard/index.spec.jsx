import React from "react";
import { shallow } from "enzyme";
import DinoCard from "./index";
import toJson from "enzyme-to-json";

const data = {
  name: "Dino name",
  image: "abelisaurus"
};

describe("<DinoCard />", () => {
  describe("default", () => {});
  const wrapper = shallow(<DinoCard dino={data} />);

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
