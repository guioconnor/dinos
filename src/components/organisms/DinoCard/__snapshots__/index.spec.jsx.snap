// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DinoCard /> should render correctly 1`] = `
<Styled(NavLink)
  to="/dino/Dino name"
>
  <styled.div>
    <styled.img
      alt="Dino name"
      src="abelisaurus"
    />
  </styled.div>
  <styled.ul>
    <DietIcon
      diet="carnivore"
    />
    <HabitatIcon />
  </styled.ul>
  <styled.h2>
    Dino name
  </styled.h2>
</Styled(NavLink)>
`;
